internal/mock/cloudsql.go

```diff
@@ -77,6 +80,72 @@ func generateInstanceCerts() (*rsa.PrivateKey, *x509.Certificate, error) {
 		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
 		BasicConstraintsValid: true,
 	}
+	certBytes, err := x509.CreateCertificate(
+		rand.Reader, cert, cert, &key.PublicKey, key)
+	if err != nil {
+		return nil, nil, tls.Certificate{}, err
+	}
+
+	caPEM := &bytes.Buffer{}
+	pem.Encode(caPEM, &pem.Block{Type: "CERTIFICATE", Bytes: certBytes})
+
+	caKeyPEM := &bytes.Buffer{}
+	pem.Encode(caKeyPEM, &pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: x509.MarshalPKCS1PrivateKey(key),
+	})
+
+	serverCert, err := tls.X509KeyPair(caPEM.Bytes(), caKeyPEM.Bytes())
+	if err != nil {
+		return nil, nil, tls.Certificate{}, err
+	}
 
-	return privKey, cert, nil
+	return key, cert, serverCert, nil
```

Comment:
Again, it's not clear to me why we have cert and serverCert.

Do we still need cert if we are generating the PEMs here? Should we just return the PEM format and use it for signing?